pkgs/by-name/ba/balatro/package.nix

```
@@ -0,0 +1,79 @@
+{
+  stdenv,
+  lib,
+  pkgs,
+  makeDesktopItem,
+  requireFile,
+  withMods ? true,
+}:
+let
+  version = "1.0.1g";
+  balatroExe = requireFile {
+    name = "Balatro-${version}.exe";
+    url = "https://store.steampowered.com/app/2379780/Balatro/";
+    # Use `nix hash file --sri --type sha256` to get the correct hash
+    hash = "sha256-X7t/T3lkLnDzewlmkxHjdePH/hlwoachuWfoUC4ntd0=";
+  };
+in
+stdenv.mkDerivation {
+  pname = "balatro";
+  inherit version;
+  nativeBuildInputs = with pkgs; [
+    p7zip
+    copyDesktopItems
+  ];
+  buildInputs = with pkgs; [ love ] ++ lib.optional withMods pkgs.lovely-injector;
+  src = ./.;
+  desktopItems = [
+    (makeDesktopItem {
+      name = "balatro";
+      desktopName = "Balatro";
+      exec = "balatro";
+      keywords = [ "Game" ];
+      categories = [ "Game" ];
+      icon = "balatro";
+    })
+  ];
+  buildPhase = ''
+    runHook preBuild
+    mkdir -p $out/share/{applications,icons}
+
+    # "Patch" game and extract icon
+    tmpdir=$(mktemp -d)
+    7z x ${balatroExe} -o$tmpdir -y
+    patch $tmpdir/globals.lua < $src/globals.patch
+    patchedExe=$(mktemp -u).zip
+    7z a $patchedExe $tmpdir/*
+    cp $tmpdir/resources/textures/2x/balatro.png $out/share/icons/balatro.png
+
+    # "Build" lovely game
+    (cat ${pkgs.love}/bin/love;  cat $patchedExe) > $out/share/Balatro
+    chmod +x $out/share/Balatro
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cat > $out/bin/balatro <<EOF
+    #!${pkgs.stdenv.shell}
+    ${
+      if withMods then "LD_PRELOAD=${pkgs.lovely-injector}/lib/liblovely.so" else ""
+    } exec $out/share/Balatro
+    EOF
+    chmod +x $out/bin/balatro
```

```suggestion
    install -Dm444 $tmpdir/resources/textures/2x/balatro.png -t $out/share/icons/

    cat ${lib.getExe love} $patchedExe > $out/share/Balatro
    
    makeWrapper $out/share/Balatro $out/bin/balatro \
      ${lib.optionalString withMods "--prefix LD_PRELOAD : '${lovely-injector}/lib/liblovely.so'"}
```

The `cat` and `LD_PRELOAD` give me pause, but this is how we'd write it most clearly. I'd have to test this out to see if there's an even simpler way to write this.

---

Both issues are related to [lovely-injector linux support](https://github.com/ethangreen-dev/lovely-injector/pull/66).

The `cat` issue is [a hack suggested by whitelje](https://github.com/ethangreen-dev/lovely-injector/pull/66#issuecomment-2319615509) to make it so that lovely will pick up `Balatro` as the game name.

The `LD_PRELOAD` bit is used to load lovely and it is the 'official' way of doing it.

---

In that case please also add comments in the code explaining this

---

> [!NOTE]
> I had to add the `chmod +x` otherwise `makeWrapper` would fail to create the wrapper, complaining the `$out/share/Balatro` file was not executable